specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json

```diff
@@ -0,0 +1,1022 @@
+          "type": "string",
+          "format": "date-time",
+          "description": "The UTC processing end time of the alert in ISO8601 format."
+        },
+        "entities": {
+          "readOnly": true,
+          "type": "array",
+          "description": "A list of entities related to the alert.",
+          "items": {
+            "$ref": "#/definitions/AlertEntity"
+          }
+        },
+        "isIncident": {
+          "readOnly": true,
+          "type": "boolean",
+          "description": "This field determines whether the alert is an incident (a compound grouping of several alerts) or a single alert."
+        },
+        "correlationKey": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Key for corelating related alerts. Alerts with the same correlation key considered to be related."
+        },
+        "extendedProperties": {
+          "$ref": "#/definitions/AlertExtendedProperties",
+          "description": "Custom properties for the alert."
+        },
+        "compromisedEntity": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The display name of the resource most related to this alert."
+        }
+      }
+    },
+    "AzureResourceIdentifier": {
+      "type": "object",
+      "description": "Azure resource identifier.",
+      "x-ms-discriminator-value": "AzureResource",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceIdentifier"
+        }
+      ],
+      "properties": {
+        "azureResourceId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "ARM resource identifier for the cloud resource being alerted on"
+        }
+      }
+    },
+    "LogAnalyticsIdentifier": {
+      "type": "object",
+      "description": "Represents a Log Analytics workspace scope identifier.",
+      "x-ms-discriminator-value": "LogAnalytics",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceIdentifier"
+        }
+      ],
+      "properties": {
+        "workspaceId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The LogAnalytics workspace id that stores this alert."
+        },
+        "workspaceSubscriptionId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The azure subscription id for the LogAnalytics workspace storing this alert.",
+          "pattern": "^[0-9A-Fa-f]{8}-([0-9A-Fa-f]{4}-){3}[0-9A-Fa-f]{12}$"
+        },
+        "workspaceResourceGroup": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The azure resource group for the LogAnalytics workspace storing this alert"
+        },
+        "agentId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "(optional) The LogAnalytics agent id reporting the event that this alert is based on."
+        }
+      }
+    },
+    "ResourceIdentifier": {
```

Comment:
Please model resource identifiers using the standard ARM id string format rather than writing a custom model for it. This has the following benefits:
1. Consistency with other ARM services.
2. Allows other services (ARM, ARG, ARM Deployments) to programatically understand resource dependencies.
3. Avoids users of the SDK to have to write resourceId parsing (e.g. if a user wants to fetch a loganalytics resource and pass the id to this field).

Reply:
Hi, can you explain " standard ARM id string format" ? 
ResourceIdentifiers has 2 formatting types, which property "type" is common , and each has its own additional properties. 
Do you mean to include only the type property as string, and allow additional properties?

Reply:
I think Anthony is talking about using the standard ARM resource ID string (something like `/subscriptions/{id}/resourceGroups/{rg}/providers...`) You are basically doing that with Azure Resources, but I definitely see the challenge with LA which is partially in the data plane (due to agent IDs).

---
In reply to: [437166909](https://github.com/Azure/azure-rest-api-specs/pull/8805#discussion_r437166909) [](ancestors = 437166909)

Reply:
We definitely should drop "Identifier" from the definition name and the property above (see comment above). It can be confused with the meaning of the `id` property which is the resource identifier.

---
In reply to: [439029078](https://github.com/Azure/azure-rest-api-specs/pull/8805#discussion_r439029078) [](ancestors = 439029078,437166909)

Reply:
See my comment on your other comment about the resourceIdentifiers. It is not just the "ARMResourceId" , there are some identifiers for the resource , ARMResourceId is identifier in terms of AzureResource. There are some more resource identifiers for the resource

Reply:
Replied on that one. I'm disagreeing with Anthony and agreeing with you, but have a concern about naming.

---
In reply to: [439064304](https://github.com/Azure/azure-rest-api-specs/pull/8805#discussion_r439064304) [](ancestors = 439064304)